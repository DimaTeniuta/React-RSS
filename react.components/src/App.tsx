import AppRouter from 'components/AppRouter';
import Footer from 'components/Footer/Footer';
import Header from 'components/Header/Header';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

function App() {
  return (
    <>
      <BrowserRouter>
        <Header />
        <AppRouter />
        <Footer />
      </BrowserRouter>
    </>
  );
}

export default App;
