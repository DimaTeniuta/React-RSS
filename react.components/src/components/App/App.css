.container {
  width: 100%;
  min-height: 100vh;
  padding: 0 0 40px 0;
  background-image: url(../../../public/assets/images/background.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  background-attachment: fixed;
}