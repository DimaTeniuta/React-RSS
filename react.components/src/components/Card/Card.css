@import '../../styles/variables.css';

.card {
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  row-gap: 15px;
  max-width: 430px;
  padding: 10px;
  backdrop-filter: var(--filter);
  background-color: var(--color-background-box);
  border: 2px solid var(--color-grey);
  border-radius: 10px;
  font-size: 16px;
}

.card__image {
  width: 400px;
  height: 280px;
  border-radius: 5px;
}

.card__wrap-description {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
}

.card__title {
  color: var(--color-white);
  font-size: 18px;
}

.card__likes {
  position: absolute;
  right: 40px;
  bottom: 30px;
  color: var(--color-white);
}

.card__likes__icon {
  position: absolute;
  right: 15px;
  bottom: 30px;
  width: 20px;
  height: 20px;
  background-image: url(../../../public/assets/svg/likes.svg);
  background-repeat: no-repeat;
  background-size: cover;
}