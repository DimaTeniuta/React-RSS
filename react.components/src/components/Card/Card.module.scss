@import '../../styles/variables.scss';

.card {
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  row-gap: 15px;
  max-width: 430px;
  padding: 10px;
  backdrop-filter: var(--filter);
  background-color: var(--color-background-box);
  border: 2px solid var(--color-grey);
  border-radius: 10px;
  font-size: 16px;

  .image {
    width: 400px;
    height: 280px;
    border-radius: 5px;
  }
  
  .title {
    color: var(--color-white);
    font-size: 18px;
  }
  
  .likes {
    position: absolute;
    right: 40px;
    bottom: 30px;
    color: var(--color-white);
  }
  
  .icon {
    position: absolute;
    right: 15px;
    bottom: 30px;
    width: 20px;
    height: 20px;
    background-image: url(../../../public/assets/svg/likes.svg);
    background-repeat: no-repeat;
    background-size: cover;
  }
}

.standardBtn {
  width: 150px;
  height: 40px;
  margin: 10px auto;
  padding: 0 25px;
  backdrop-filter: var(--filter);
  background-color: var(--color-background-box);
  border: 1px solid var(--color-blue);
  border-radius: 20px;
  color: var(--color-white);
  font-family: sans-serif;
  font-size: var(--font-regular);
  font-weight: 400;
  cursor: pointer;
  transition: 0.3s;

  &:hover {
    cursor: pointer;
    color: #e7da6c;
  }
}

