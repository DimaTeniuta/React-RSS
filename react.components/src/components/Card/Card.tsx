import Button from 'components/UI/Button/Button';
import React from 'react';
import { ResultsData } from 'types/generalTypes';
import classes from './Card.module.scss';
import btnClasses from '../UI/Button/Button.module.scss';

type DataCard = {
  data: ResultsData;
};

export const Card: React.FC<DataCard> = (props) => {
  const data = props.data;
  return (
    <div className={classes.card} data-testid="test-card">
      <img className={classes.images} src={data.urls?.small} alt="card-image" />
      <p className={classes.title}>{data.description ? data.description : 'Unknown'}</p>
      <span className={classes.likes}>{data.likes}</span>
      <div className={classes.icon}></div>
      <form action={data.urls?.full} target="_blank">
        <Button className={btnClasses.standardBtn}>Full size</Button>
      </form>
    </div>
  );
};
