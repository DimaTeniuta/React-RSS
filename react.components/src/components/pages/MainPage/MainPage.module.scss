.container {
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  row-gap: 30px;
  padding: 30px;

  .wrapCards {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    gap: 20px;
  }
}

.standardBtn {
  width: 150px;
  height: 40px;
  margin: 10px auto;
  padding: 0 25px;
  backdrop-filter: var(--filter);
  background-color: var(--color-background-box);
  border: 1px solid var(--color-blue);
  border-radius: 20px;
  color: var(--color-white);
  font-family: sans-serif;
  font-size: var(--font-regular);
  font-weight: 400;
  cursor: pointer;
  transition: 0.3s;

  &:hover {
    cursor: pointer;
    color: #e7da6c;
  }
}