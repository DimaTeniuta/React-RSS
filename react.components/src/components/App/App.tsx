import AppRouter from 'components/AppRouter';
import Footer from 'components/Footer/Footer';
import Header from 'components/Header/Header';
import React from 'react';
import './App.css';

function App() {
  return (
    <div className="container">
      <Header />
      <AppRouter />
      <Footer />
    </div>
  );
}

export default App;
