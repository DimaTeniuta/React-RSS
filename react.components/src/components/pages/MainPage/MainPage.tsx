import TextWindow from 'components/TextWindow/TextWindow';
import Search from 'components/Search/Search';
import React, { useContext, useState } from 'react';
import classes from './MainPage.module.scss';
import Card from 'components/Card/Card';
import Loader from 'components/Loader/Loader';
import { MainContext } from 'context/MainProvider/MainProvider';

const MainPage = (): JSX.Element => {
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const { data } = useContext(MainContext);

  const toggleLoader = (): void => {
    setIsLoading((value) => !value);
  };

  return (
    <div className={classes.container} data-testid="mainPage">
      <Search toggleLoader={toggleLoader} />

      <div className={classes.wrapCards}>
        {isLoading ? (
          <Loader />
        ) : data.length ? (
          data.map((el) => <Card key={el.id} data={el} />)
        ) : (
          <TextWindow title="Not Found">Try entering another query</TextWindow>
        )}
      </div>
    </div>
  );
};

export default MainPage;
