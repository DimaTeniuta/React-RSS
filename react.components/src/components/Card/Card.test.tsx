import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from 'App';
import MainPage from 'components/pages/MainPage/MainPage';
import React from 'react';
import Card from './Card';

const mockData = {
  id: '1',
  description: 'Test',
  likes: 286,
  urls: {
    full: 'https://hd.unsplash.com/photo-1416339306562-f3d12fefd36f',
    small:
      'https://images.unsplash.com/photo-1416339306562-f3d12fefd36f?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&s=263af33585f9d32af39d165b000845eb',
  },
};

describe('Card', () => {
  it('Card renders', () => {
    render(<Card data={mockData} />);
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.getByText(/more details/i)).toBeInTheDocument();
    expect(screen.getByText(mockData.description)).toBeInTheDocument();
    expect(screen.queryByTestId('modalWindowWrap')).not.toBeInTheDocument();
  });

  it('render 30 cards', async () => {
    render(<App />);
    const main = screen.getByTestId('mainLink');
    userEvent.click(main);
    const cards = await screen.findAllByTestId('test-card');
    expect(cards.length).toBe(10);
  });

  it('Cards snapshot', () => {
    const cards = render(<MainPage />);
    expect(cards).toMatchSnapshot();
  });
});
