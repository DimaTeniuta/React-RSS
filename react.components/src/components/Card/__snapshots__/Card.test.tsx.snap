// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
        data-testid="test-card"
      >
        <img
          alt="card-image"
          class="card__image"
          src="https://images.unsplash.com/photo-1416339306562-f3d12fefd36f?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&s=263af33585f9d32af39d165b000845eb"
        />
        <p
          class="card__title"
        >
          Test
        </p>
        <span
          class="card__likes"
        >
          286
        </span>
        <div
          class="card__likes__icon"
        />
        <form
          action="https://hd.unsplash.com/photo-1416339306562-f3d12fefd36f"
          target="_blank"
        >
          <button
            class="btn"
          >
            Full size
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
      data-testid="test-card"
    >
      <img
        alt="card-image"
        class="card__image"
        src="https://images.unsplash.com/photo-1416339306562-f3d12fefd36f?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&s=263af33585f9d32af39d165b000845eb"
      />
      <p
        class="card__title"
      >
        Test
      </p>
      <span
        class="card__likes"
      >
        286
      </span>
      <div
        class="card__likes__icon"
      />
      <form
        action="https://hd.unsplash.com/photo-1416339306562-f3d12fefd36f"
        target="_blank"
      >
        <button
          class="btn"
        >
          Full size
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card Cards snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main__container"
        data-testid="main-page"
      >
        <div
          class="main__wrap-input"
        >
          <div>
            <input
              class="my-input"
              placeholder="Search"
              type="text"
              value=""
            />
            <span
              class="main__input-icon-close"
              data-testid="clearBtn"
            />
          </div>
          <button
            class="btn"
            data-testid="test-search-btn"
          >
            Search
          </button>
        </div>
        <div
          class="main__wrap-cards"
        >
          <div
            class="error__wrap"
            data-testid="error-page"
          >
            <h1
              class="error__title"
            >
              Not Found
            </h1>
            <p
              class="error__message"
            >
              Try entering another query
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main__container"
      data-testid="main-page"
    >
      <div
        class="main__wrap-input"
      >
        <div>
          <input
            class="my-input"
            placeholder="Search"
            type="text"
            value=""
          />
          <span
            class="main__input-icon-close"
            data-testid="clearBtn"
          />
        </div>
        <button
          class="btn"
          data-testid="test-search-btn"
        >
          Search
        </button>
      </div>
      <div
        class="main__wrap-cards"
      >
        <div
          class="error__wrap"
          data-testid="error-page"
        >
          <h1
            class="error__title"
          >
            Not Found
          </h1>
          <p
            class="error__message"
          >
            Try entering another query
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
